13/realnitinworks/directors.py

```diff
@@ -0,0 +1,118 @@
+from collections import namedtuple, Counter, defaultdict
+import csv
+import statistics
+
+MOVIES_DATASET = "movie_metadata.csv"
+FROM_YEAR = "1960"
+MIN_MOVIES = 4
+TOP_N = 20
+
+Movie = namedtuple('Movie', 'director title year rating')
+
+
+def convert_csv_to_dict(data=MOVIES_DATASET):
+    """ Yield Movie(director, title, year, rating) record from input csv file """
+    with open(data) as csvfile:
+        for movie in csv.DictReader(csvfile):
+            yield Movie(
+                director=movie['director_name'],
+                title=movie['movie_title'].strip(),
+                year=movie['title_year'],
+                rating=movie['imdb_score']
+            )
+
+
+def from_year(movies, year=FROM_YEAR):
+    """ Filter movies based on year"""
+    for movie in movies:
+        if not year:
+            continue
+        if movie.year >= year:
+            yield movie
+
+
+def director_filter(movies, min_movies=MIN_MOVIES):
+    """ Filter movies whose their director's min movies """
+    director_count = Counter(
+        movie.director
+        for movie in movies
+    )
+
+    movies = from_year(convert_csv_to_dict())
+    return (
+        movie
+        for movie in movies
+        if director_count[movie.director] >= min_movies
+    )
+
+
+def map_director_movie(movies):
+    """ Make a mapping of directors and their movies"""
+    def by_rating(movie):
+        return movie.rating
+
+    directors = defaultdict(lambda: defaultdict(list))
+    for movie in sorted(movies, key=by_rating, reverse=True):
+        directors[movie.director]["movies"].append(movie)
+
+    return directors
+
+
+def calculate_mean_imdb_score(director_movie_map):
+    """ Add mean_rating key into the input map"""
+    for director, value in director_movie_map.items():
+        imdb_scores = []
+        for movie in value["movies"]:
+            imdb_scores.append(float(movie.rating))
+        director_movie_map[director]["mean_rating"] = round(statistics.mean(imdb_scores), 1)
```

Comment:
could keep it smaller by saving `round...` into a variable and assign the variable